homeassistant/components/logbook/websocket_api.py

```diff
@@ -0,0 +1,266 @@
+        entity_ids,
+        device_ids,
+        None,
+        timestamp=True,
+        include_entity_name=False,
+    )
+
+    stream_queue: asyncio.Queue[Event] = asyncio.Queue(MAX_PENDING_LOGBOOK_EVENTS)
+    subscriptions: list[CALLBACK_TYPE] = []
+
+    @callback
+    def _forward_events(event: Event) -> None:
+        stream_queue.put_nowait(event)
+
+    for event_type in event_types:
+        subscriptions.append(
+            hass.bus.async_listen(event_type, _forward_events, run_immediately=True)
+        )
+    if entity_ids:
+        subscriptions.append(
+            async_track_state_change_event(hass, entity_ids, _forward_events)
+        )
+    else:
+        # We want the firehose
+        subscriptions.append(
+            hass.bus.async_listen(
+                EVENT_STATE_CHANGED, _forward_events, run_immediately=True
+            )
+        )
+
+    message, last_time = await get_instance(hass).async_add_executor_job(
+        _ws_formatted_get_events,
+        msg["id"],
+        start_time,
+        utc_now,
+        messages.event_message,
+        event_processor,
+    )
+    task = asyncio.create_task(
+        async_stream_events(
+            hass,
+            connection,
+            msg["id"],
+            stream_queue,
+            event_processor,
+            last_time or start_time,
+        )
+    )
+
+    def _unsub() -> None:
+        """Unsubscribe from all events."""
+        for subscription in subscriptions:
+            subscription()
+        task.cancel()
+
+    connection.subscriptions[msg["id"]] = _unsub
+    connection.send_result(msg["id"])
+    connection.send_message(message)
+
+
+@websocket_api.websocket_command(
+    {
+        vol.Required("type"): "logbook/get_events",
+        vol.Required("start_time"): str,
+        vol.Optional("end_time"): str,
+        vol.Optional("entity_ids"): [str],
+        vol.Optional("device_ids"): [str],
+        vol.Optional("context_id"): str,
+    }
+)
+@websocket_api.async_response
+async def ws_get_events(
+    hass: HomeAssistant, connection: websocket_api.ActiveConnection, msg: dict
+) -> None:
+    """Handle logbook get events websocket command."""
+    start_time_str = msg["start_time"]
+    end_time_str = msg.get("end_time")
+    utc_now = dt_util.utcnow()
+
+    if start_time := dt_util.parse_datetime(start_time_str):
+        start_time = dt_util.as_utc(start_time)
+    else:
+        connection.send_error(msg["id"], "invalid_start_time", "Invalid start_time")
+        return
+
+    if not end_time_str:
+        end_time = utc_now
+    elif parsed_end_time := dt_util.parse_datetime(end_time_str):
+        end_time = dt_util.as_utc(parsed_end_time)
+    else:
+        connection.send_error(msg["id"], "invalid_end_time", "Invalid end_time")
+        return
+
+    if start_time > utc_now:
+        connection.send_result(msg["id"], [])
+        return
+
+    device_ids = msg.get("device_ids")
+    entity_ids = msg.get("entity_ids")
```

Comment:
We should filter out entities that have UOM or state class here for the existing API

Reply:
We should reduce the entity list before calling async_determine_event_types

Reply:
195f5c1a8c8836c624d9353ed6de5f4836cc6462